content_scripts/link_hints.coffee

```diff
@@ -125,6 +125,22 @@ LinkHints =
     marker
 
   #
+  # Find elements that have delegated onclick event listener assigned,
+  # and mark them with `vimium-has-delegated-onclick-listener` attribute.
+  # This will make sure those elements are discovered during `getVisibleClickableElements` call.
+  #
+  markElementsThatHaveDelegatedOnClickListener: ->
+    for element in document.querySelectorAll("*[vimium-jquery-delegated-events-selectors]")
+      selectorsStr = element.getAttribute "vimium-jquery-delegated-events-selectors"
+      continue unless selectorsStr
+
+      selectors = selectorsStr.split("|").filter (x) -> !!x
+
+      for selector in selectors
+        for child in element.querySelectorAll(selector)
```

Comment:
I have been running this for a while, it is wrong to use `querySelectorAll` here. Selectors were extracted from jQuery's `.on()` function, and jQuery supports more kinds of selectors than `querySelectorAll` (it uses [Sizzle](https://github.com/jquery/sizzle/wiki) in some cases). 

An example I found on trello.com is the following selector: `a[href=#]`.
jQuery returns some links, while `querySelectorAll` throws an exception.

Now, it sounds safe to use jQuery defined on a page &mdash; it is definitely defined as we reach this function, we don't bother if jQuery treats same selectors differently in different versions, etc.

But it seems like we cannot access jQuery, defined on the page, from the content script... Is that right, @mrmr1993?

If that's true, what would be the best way to handle this situation? 
I can easily think of the following approaches:

* Catch `querySelectorAll`'s exceptions and simply ignore those selectors.
* Introduce jQuery (or Sizzle) as a part of vimium distribution.

But I like neither of them :smile: